MarsRover/MapParser.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MarsRover
+{
+    public static class MapParser
+    {
+        public static Map ParseMap(string input)

```

Comment:
would be good to have some basic checks for validity e.g. if one of the lines is a different length